### DISCUSSION TITLE
Does plausible cull / aggregate old data to save space?

### DISCUSSION THREAD

**Discussion by binaryfire** [plausible/analytics#856] [Anything Else]
Hi all

I was wondering if Plausible automatically aggregates old data (eg. just stores monthy figures for past years and drops daily data data points) to save on storage? 

Cheers

**Comment by Vigasaurus** [comment]
It doesn't, but clickhouse is incredibly efficient as a event store. For some context, my self hosted instance with almost 6 million events and 300k sessions (over 4 months) is only using about 67 MiB, so take from that what you will.

**Reply by binaryfire** [reply]
Nice, that's pretty good. But I still think it'd be great to have a way to prune old data.  We've got some clients with 1mil+ sessions a month. Long term storage is going to become problematic at some point.

**Reply by ukutaht** [reply]
If you're willing to get your hands dirty with Clickhouse you can add a TTL clause to your tables: https://clickhouse.tech/docs/en/sql-reference/statements/alter/ttl/

**Reply by binaryfire** [reply]
Cool thanks, I'll take a look. But I'm more of a frontend guy so not the best with SQL :P. Anyway, just putting it out there for the team... A script users could run to aggregate old data would be quite useful for people running self-hosted installs.
